### PR TITLE
Remove EDG 3.2 workaround

### PR DESCRIPTION
**Pull request by Kojoley**
EDG 3.2 is pretty old (the workaround added 16 years ago).

### CODE REVIEW

**Comment by apolukhin** [comment]
I think it should be the other way around: all the enable_if must be removed in favor of the `typename Visitor::type`

**Comment by Kojoley** [comment]
Hmm, I though it was made to allow `boost::apply_visitor<Visitor const>(...)`.

File: include/boost/variant/detail/apply_visitor_binary.hpp [hunk 1]
```diff
@@ -14,16 +14,12 @@
...
 
 #include <boost/config.hpp>
-#include <boost/detail/workaround.hpp>
 
 #include <boost/variant/detail/apply_visitor_unary.hpp>
 
 #include <boost/utility/enable_if.hpp>
-
-#if BOOST_WORKAROUND(__EDG__, BOOST_TESTED_AT(302))
 #include <boost/mpl/not.hpp>
```
Comment (apolukhin) [inline include/boost/variant/detail/apply_visitor_binary.hpp]: Please, also drop the unnecessary headers.

File: include/boost/variant/detail/apply_visitor_unary.hpp [hunk 2]
```diff
@@ -14,15 +14,12 @@
 #define BOOST_VARIANT_DETAIL_APPLY_VISITOR_UNARY_HPP
 
 #include <boost/config.hpp>
-#include <boost/detail/workaround.hpp>
 #include <boost/move/utility.hpp>
 
-#if BOOST_WORKAROUND(__EDG__, BOOST_TESTED_AT(302))
 #include <boost/core/enable_if.hpp>
```
Comment (apolukhin) [inline include/boost/variant/detail/apply_visitor_unary.hpp]: Drop those headers too

**Comment by apolukhin** [comment]
Awesome! Many thanks!

**Pull request merged**